pyproject.toml

```diff
@@ -27,6 +27,7 @@ dependencies = [
     "tenacity >= 8",
     "importlib-resources >= 6.1.1; python_version < '3.9'",
     "multiprocess",
+    "dill >= 0.3.7",
```

Comment:
I'm not a big fan of `dill`, `pickle` and similar, so maybe we can explore other ways for saving/dumping the `Task` configuration?

Reply:
As discussed internally, will review this in the future😄